src/userLayout/payment/Checkout.tsx

```diff
@@ -0,0 +1,163 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { Box, Grid, Typography, Button } from "@mui/material";
+import {
+  CardElement,
+  useStripe,
+  useElements,
+  AddressElement,
+} from "@stripe/react-stripe-js";
+import { useContext, useEffect, useState } from "react";
+import axios from "axios";
+import { toast } from "react-toastify";
+import { AuthContext } from "../../Context/Components/AuthContext";
+import { useNavigate, useParams} from "react-router-dom";
+
+const Checkout = () => {
+  const stripe = useStripe();
+  const elements = useElements();
+  const navigate = useNavigate();
+  const [bookingDetails, setBookingDetails] = useState({});
+  console.log(bookingDetails)
+  const { id } = useParams();
+  // const id ='65a9968ea5d9953dd42d11aa'
+  const handleSubmit = async (event: any) => {
+    // We don't want to let default form submission happen here,
+    // which would refresh the page.
+    event.preventDefault();
+
+    if (!stripe || !elements) {
+      // Stripe.js hasn't yet loaded.
+      // Make sure to disable form submission until Stripe.js has loaded.
+      return;
+    }
+    const cardElement = elements?.getElement(CardElement);
+    console.log(cardElement);
+    if (!cardElement) {
+      console.error("Card element not found");
+      return;
+    }
+    const { error, token } = await stripe.createToken(cardElement);
+    console.log(token);
+    const tokenId = token?.id;
+
+    const payBooking = async (id: string) => {
+      await axios
+        .post(
+          `${baseUrl}/v0/portal/booking/${id}/pay`,
+          { token: tokenId },
+          {
+            headers: {
+              Authorization: `Bearer ${loginData}`,
+            },
+          }
+        )
+        .then((response) => {
```

Comment:
Why not use await instead, it is not good to mix between then catch and await.  We should also wrap the first part into a try block and make a catch block to catch the errors.

The form is currently always run the then block even if the form is empty and there are errors.